### PR TITLE
Feature/parts list post ids

### LINKED ISSUE #167: URL limit exceeded when doing a GET containing too many part ids
How to reproduce:

1. Have a lot of dependencies registered
2. Without filtering, click select all
3. Click on Export Selected

### PR DESCRIPTION
**Pull request by lloydmeta**
Closes #167

### CODE REVIEW

File: app/controllers/support/PartListFilterSupport.scala [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+  val partListFilterForm: Form[PartListFilter] = Form(
+    mapping(
+      "ids" -> seq(nonEmptyText)
+    )(PartListFilter.apply)(PartListFilter.unapply)
+  )
+
+}
+
+case class PartListFilter(@BeanProperty ids: Seq[String])
```
Comment (mauhiz) [inline app/controllers/support/PartListFilterSupport.scala]: Why do you need a `BeanProperty` here?

File: test/controllers/PartsControllerSpec.scala [hunk 2]
```diff
@@ -96,4 +96,13 @@ class PartsControllerSpec extends FlatSpec with Matchers with MockitoSugar with
...
   }
 
+  it should "show a filtered list of octoparts when POST is used" in {
+    val result = controller.listPost.apply(FakeRequest().withJsonBody(Json.obj("ids" -> Json.arr("void"))))
+
+    status(result) should be(200)
+    contentAsString(result) should include("void")
+    contentAsString(result) shouldNot include("error")
+    contentAsString(result) shouldNot include("slow")
+  }
```
Comment (mauhiz) [inline test/controllers/PartsControllerSpec.scala]: What happens if you post without a filter? Does it behave like the GET?

File: app/controllers/support/PartListFilterSupport.scala (hunk 1 above)
Comment (lloydmeta) [inline app/controllers/support/PartListFilterSupport.scala]: I thought it was required for Swagger to pick up the param, but it seems to work without. Removed.

File: test/controllers/PartsControllerSpec.scala (hunk 2 above)
Comment (lloydmeta) [inline test/controllers/PartsControllerSpec.scala]: Yes, if you POST without a filter, it behaves the same way the GET would; no filter is applied.

**Pull request merged**